src/serde_arrow_type.erl

```diff
@@ -0,0 +1,58 @@
+%%%-----------------------------------------------------------------------------
+%% @doc The `serde_arrow_type' module provides types in order to represent Apache
+%% Arrow specific primitive logical datatypes, as well as Erlang datatypes which
+%% `serde_arrow' supports serializing and deserializing.
+%% @end
+%%%-----------------------------------------------------------------------------
+-module(serde_arrow_type).
+-export_type([
+    arrow_type/0,
+    arrow_bool/0,
+    arrow_int/0,
+    arrow_uint/0,
+    arrow_float/0,
+    erlang_type/0
+]).
+
+-type arrow_type() ::
+    arrow_bool()
+    | arrow_int()
+    | arrow_uint()
+    | arrow_float().
+%% Any primitive logical type in Apache Arrow that is supported by `serde_arrow'.
+
+-type arrow_bool() ::
+    arrow_true
+    | arrow_false.
```

Comment:
Does Arrow have `true` and `false` as types? Because if the type is simply boolean, then this should probably be the atom `boolean`. In other words, someone would say this:

    serialize(MyList, boolean)

And not `serialize(MyList, arrow_boolean)` or `serialize(MyList, arrow_true)`, right?

Reply:
`boolean()` is a built-in type in Erlang, which is why I named it `arrow_boolean`. `true` as you know, is Erlang's `true`, which is why I use the atom `arrow_true`.

Reply:
I understand but I think calling it `arrow_true` means you will specify a buffer to have type `arrow_true`, which doesn’t make sense. You will most likely say a buffer has type `boolean` and inside of the buffer you have `boolean()` values. Like a `{s, 32}` buffer hás `integer` values.

Reply:
You can leave it as is, if it is wrong it will be apparent later. :)